pallets/utxo/src/lib.rs

```diff
@@ -841,10 +820,31 @@ pub mod pallet {
             value: Value,
             address: Vec<u8>,
         ) -> DispatchResultWithPostInfo {
-            use chainscript::Script;
-
+            let (_, data, _) = bech32::decode(&address).or_else(|e| match e {
+                bech32::Error::InvalidLength => Err(DispatchError::Other(
+                    "Failed to decode address: invalid length",
+                )),
+                bech32::Error::InvalidChar(_) => Err(DispatchError::Other(
+                    "Failed to decode address: invalid character",
+                )),
+                bech32::Error::MixedCase => {
+                    Err(DispatchError::Other("Failed to decode address: mixed case"))
+                }
+                bech32::Error::InvalidChecksum => Err(DispatchError::Other(
+                    "Failed to decode address: invalid checksum",
+                )),
+                _ => Err(DispatchError::Other("Failed to decode address")),
+            })?;
+
+            let mut tmp: &[u8] = &Vec::<u8>::from_base32(&data)
+                .or_else(|_| Err(DispatchError::Other("Base32 conversion failed")))?;
+
+            let dest: Destination<T::AccountId> = Destination::decode(&mut tmp).or_else(|_| {
+                Err(DispatchError::Other(
+                    "Failed to decode buffer into `Destination`",
+                ))
+            })?;
```

Comment:
Minor possible improvement (haven't tried, might be wrong):

Given `something: Result<_, _>`, we can replace the following pattern:
```rust
soething.or_else(|_| Err(foo))
```
with
```rust
something.map_err(|_| foo)
```